app/services/view/HolderOfAuthorisationCodes.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.view
+
+import java.util.Locale
+
+import connectors.CodeListConnector
+import javax.inject.{Inject, Singleton}
+import models.codes.HolderOfAuthorisationCode
+
+@Singleton
+class HolderOfAuthorisationCodes @Inject()(codeListConnector: CodeListConnector) {
+
+  private def description(h: HolderOfAuthorisationCode): String = s"${h.code} - ${h.description}"
+
+  def getCodeDescription(locale: Locale, code: String): String =
+    codeListConnector.getHolderOfAuthorisationCodes(locale).get(code).fold("")(description)
+
+  def asListOfAutoCompleteItems(locale: Locale): List[AutoCompleteItem] =
+    codeListConnector.getHolderOfAuthorisationCodes(locale).values.map(h => AutoCompleteItem(description(h), h.code)).toList
```

Comment:
I don't agree with moving the responsibility of creating `AutoCompleteItem`s out of the existing `AutoCompleteItem` class. The method `asListOfAutoCompleteItems` is unnecessarily coupling this service class to a UI form element IMO.

Reply:
In turn, I disagree. Please, read my comment. This service class is a **UI** service class like `AutoCompleteItems` but it only has the responsability for `HolderOfAuthorisationCodes` and not for any other codes like instead has `AutoCompleteItems` which follows in my opinion a really bad design.